qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+            assert pulse.measure_channel(0) == pulse.MeasureChannel(0)
+
+    .. note:: Requires the active builder context to have a backend set.
+    """
+    return active_backend().configuration().measure(qubit)
+
+
+def acquire_channel(qubit: int) -> channels.AcquireChannel:
+    """Return ``AcquireChannel`` for ``qubit`` on the active builder backend.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            assert pulse.acquire_channel(0) == pulse.AcquireChannel(0)
+
+    .. note:: Requires the active builder context to have a backend set.
+    """
+    return active_backend().configuration().acquire(qubit)
+
+
+def control_channels(*qubits: Iterable[int]) -> List[channels.ControlChannel]:
+    """Return ``AcquireChannel`` for ``qubit`` on the active builder backend.
+
+    Return the secondary drive channel for the given qubit -- typically
+    utilized for controlling multi-qubit interactions.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+        with pulse.build(backend):
+            assert pulse.control_channels(0, 1) == [pulse.ControlChannel(0)]
+
+    .. note:: Requires the active builder context to have a backend set.
+
+    Args:
+      qubits: Tuple or list of ordered qubits of the form
+        `(control_qubit, target_qubit)`.
+
+    Returns:
+        List of control channels associated with the supplied ordered list
+        of qubits.
+    """
+    return active_backend().configuration().control(qubits=qubits)
+
+
+# Base Instructions
+def delay(duration: int,
+          channel: channels.Channel):
+    """Delay on a ``channel`` for a ``duration``.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.delay(10, d0)
+
+    Args:
+        duration: Number of cycles to delay for on ``channel``.
+        channel: Channel to delay on.
+    """
+    append_instruction(instructions.Delay(duration, channel))
+
+
+def play(pulse: Union[pulse_lib.Pulse, np.ndarray],
+         channel: channels.PulseChannel):
+    """Play a ``pulse`` on a ``channel``.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.play(pulse.Constant(10, 1.0), d0)
+
+    Args:
+        pulse: Pulse to play.
+        channel: Channel to play pulse on.
+    """
+
+    if not isinstance(pulse, pulse_lib.Pulse):
+        pulse = pulse_lib.SamplePulse(pulse)
+
+    append_instruction(instructions.Play(pulse, channel))
+
+
+def acquire(duration: int,
+            qubit_or_channel: Union[int, channels.AcquireChannel],
```

Comment:
Why accept `AcquireChannel` at all? It's redundant information

Reply:
We can match this with the `measure` function, which doesn't take a `MeasureChannel`

Reply:
This is for a transition path. It is also important with regards to multiplexing or generic signal channels.